programs/stake/src/stake_state.rs

```diff
@@ -610,82 +638,59 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
         }
 
         if let StakeState::Uninitialized = split.state()? {
-            // verify enough account lamports
-            if lamports > self.lamports()? {
-                return Err(InstructionError::InsufficientFunds);
-            }
```

Comment:
nits: how about restoring this not-duplicate check to avoid the scary `.unwrap()` (https://github.com/solana-labs/solana/pull/22663/files?diff=split&w=1#r802645422 and https://github.com/solana-labs/solana/pull/22663/files?diff=split&w=1#r796146060)

1. call `validate_split_amount()` in this `fn` from both `match` arms for `StakeState::Stake` and `StakeState::Initialize` respectively.
2. for that adjust the signature like this `fn validate_split_amount(self, split, lamports, meta)`.
3. in turn, we can remove the `match` in the `validate_split_amount` to avoid `.unwrap()` there as well.
4. profit with type safety. :)

Reply:
Love it. Will do.

Reply:
Done!